### ISSUE TITLE
[a11y]: ContentSwitcher [icon only with layer & with layer] - context of text is not identified semantically

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13980]
### Package

@carbon/react

### Browser
Safari & FF & iPhone
### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology
JAWS & VO

### Description

"The following text's in variant With Layer: 
Layer 1, Layer 2, Layer 3 "

The context of the mentioned text is not identified Semantically.

### WCAG 2.1 Violation
n/a

### Reproduction/example

Record `Number 50` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

The mentioned text is identified by the screen reader without any semantics. Hence the user may not know its purpose.

**_If_** the mentioned text identifies the content in the tab panel, these should be given appropriate heading tags. If the text identifies the tab group, These can be given as the label for the tablist instead.

This needs to be confirmed / refined by the team

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by ljcarot** [comment]
@andreancardona Why is this labeled a severity 3 when Ram says it's just an observation?

**Comment by tw15egan** [comment]
The `with layer` stories are there to ensure no visual regressions via Percy with our layer token sets and do not simulate a real-world example. No action needs to be taken here.
